ibodata/lateral_profile.py

```diff
@@ -120,7 +120,24 @@ def normalize(self, dt, allow_cast=True):
         mid = self.x_at_y(a) + w / 2.0
         self.x -= mid
 
-        ave = np.average(self.y[np.fabs(self.x) <= dt])
+        slice_y = self.y[np.fabs(self.x) <= dt]
+        slice_x = self.x[np.fabs(self.x) <= dt]
+
+        """
+        if there's no point on the edge normalization is not perfectly accurate 
+        and we are normalizing over smaller area than [-dt, +dt]
+        That's why we interpolate points on the edge below.
+        """
+        if np.argwhere(self.x == -dt).size == 0:
+            slice_y = np.append(slice_y, self.y_at_x(dt))
```

Comment:
This is expensive in terms of memory allocation. I'd avoid it.

Why not calculate first area over usuall slices.

Then do such if statements as here and add missing area, using code like:

```
if left_point_not_found:
  x_coords = (-dt, slice_x[0])
  y_coords = (self.y_at_x(-dt), slice_y[0])
  area += np.trapz(x_coords, y_coords)
```